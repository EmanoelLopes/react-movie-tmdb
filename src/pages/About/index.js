import React from 'react';

const About = () => (
  <React.Fragment>
    <h1>About</h1>
  </React.Fragment>
);

export default About;

