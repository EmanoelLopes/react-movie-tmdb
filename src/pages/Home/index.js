import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFilm, faTv } from '@fortawesome/free-solid-svg-icons';
import {
  MoviesContainer,
  TVSeriesContainer,
  HeroSectionContainer,
} from 'containers';
import { H1 } from 'utils/styled';

const Home = () => {
  return (
    <React.Fragment>
      <HeroSectionContainer />
      <div className="container">
        <H1>Popular Movies <FontAwesomeIcon icon={faFilm} /></H1>
        <MoviesContainer />
      </div>
      <div className="container">
        <H1>Popular TV Series <FontAwesomeIcon icon={faTv} /></H1>
        <TVSeriesContainer />
      </div>
    </React.Fragment>
  );
};

export default Home;
