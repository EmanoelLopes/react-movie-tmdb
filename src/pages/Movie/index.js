import React from 'react';
import { H1 } from 'utils/styled';

const Movie = () => {
  return (
    <div className="movie">
      <div className="container">
        <H1>Movie</H1>
      </div>
    </div>
  );
};

export default Movie;
