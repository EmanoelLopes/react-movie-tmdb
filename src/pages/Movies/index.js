import React from 'react';
import intl from 'react-intl-universal';
import { Switch, Route } from 'react-router-dom';
import SubNav from 'components/SubNav';
import { MoviesContainer } from 'containers';
import { PageWrapper, Container, H1 } from 'utils/styled';
import { endpoints } from 'services';

const {
  topRated,
  popular,
  upComing,
  nowPlaying,
} = endpoints.movie;

const TopRated = () => {
  return (
    <Container>
      <MoviesContainer type={topRated} />
    </Container>
  );
};

const Popular = () => {
  return (
    <Container>
      <MoviesContainer type={popular} />
    </Container>
  );
};

const NowPlaying = () => {
  return (
    <Container>
      <MoviesContainer type={nowPlaying} />
    </Container>
  );
};

const UpComing = () => {
  return (
    <Container>
      <MoviesContainer type={upComing} />
    </Container>
  );
};

const Movies = () => {
  const items = [
    { link: '/movies/top-rated', page: `${intl.get('SUBNAV.MOVIES.TOP_RATED')}` },
    { link: '/movies/popular', page: `${intl.get('SUBNAV.MOVIES.POPULAR')}` },
    { link: '/movies/now-playing', page: `${intl.get('SUBNAV.MOVIES.NOW_PLAYING')}` },
    { link: '/movies/up-coming', page: `${intl.get('SUBNAV.MOVIES.UP_COMING')}` },
  ];
  
  return (
    <PageWrapper>
      <Container>
        <H1>{intl.get('MOVIES_PAGE.TITLE')}</H1>
        <SubNav items={items} />
      </Container>
      <Switch>
        <Route exact path="/movies" component={TopRated} />
        <Route exact path="/movies/top-rated" component={TopRated} />
        <Route path="/movies/popular" component={Popular} />
        <Route path="/movies/up-coming" component={UpComing} />
        <Route path="/movies/now-playing" component={NowPlaying} />
      </Switch>
    </PageWrapper>
  );
};

export default Movies;
