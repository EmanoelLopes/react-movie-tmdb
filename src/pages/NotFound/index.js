import React from 'react';

const NotFound = () => (
  <React.Fragment>
    <h1>Not Found Page :/</h1>
  </React.Fragment>
);

export default NotFound;
