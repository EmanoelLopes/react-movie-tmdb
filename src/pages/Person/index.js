import React from 'react';

const Person = () => (
  <div className="person">
    <h1>Person</h1>
  </div>
);

export default Person;
