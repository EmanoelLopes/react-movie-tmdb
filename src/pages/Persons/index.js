import React from 'react';

const Persons = () => (
  <div className="persons">
    <h1>Persons</h1>
  </div>
);

export default Persons;
