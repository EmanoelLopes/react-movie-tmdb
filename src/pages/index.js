import { default as Home } from 'pages/Home';
import { default as About } from 'pages/About';
import { default as TVs } from 'pages/TVs';
import { default as TV } from 'pages/TV';
import { default as Movies } from 'pages/Movies';
import { default as Movie } from 'pages/Movies';
import { default as Persons } from 'pages/Persons';
import { default as Person } from 'pages/Person';
import { default as NotFound } from 'pages/NotFound';

export {
  Home,
  About,
  TVs,
  TV,
  Movies,
  Movie,
  Persons,
  Person,
  NotFound,
};
