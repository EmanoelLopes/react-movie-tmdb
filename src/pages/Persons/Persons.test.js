import React from 'react';
import { mount } from 'enzyme';
import { Persons } from 'pages';
import Root from 'components/Root';

let wrapped;
let defaultProps = {};

beforeEach(() => {
  wrapped = mount(
    <Root>
      <Persons {...defaultProps} />
    </Root>
  );
});

afterEach(() => {
  wrapped.unmount();
});

describe('<Persons />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
