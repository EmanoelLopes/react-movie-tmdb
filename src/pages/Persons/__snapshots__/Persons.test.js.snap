// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Persons /> should match snapshot 1`] = `
<Root>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Persons>
      <div
        className="persons"
      >
        <h1>
          Persons
        </h1>
      </div>
    </Persons>
  </Provider>
</Root>
`;
